### PR TITLE
switch grammar_search_score_function to hff_lookahead_depth1

### PR DESCRIPTION
**Pull request by ronuchit**
now with fancy regex

### CODE REVIEW

File: src/approaches/grammar_search_invention_approach.py [hunk 1]
```diff
@@ -470,23 +471,13 @@ def _create_score_function(
     if score_function_name == "hadd_match":
         return _RelaxationHeuristicMatchBasedScoreFunction(
             initial_predicates, atom_dataset, train_tasks, candidates, "hadd")
-    if score_function_name == "hadd_lookahead":
+    match = re.match(r"(\w+)_lookahead_depth(\d+)", score_function_name)
```
Comment (tomsilver) [inline src/approaches/grammar_search_invention_approach.py:474]: add a comment describing what values are supported

File: src/settings.py [hunk 2]
```diff
@@ -120,7 +120,7 @@ class GlobalSettings:
     grammar_search_pred_complexity_weight = 1
     grammar_search_max_predicates = 50
     grammar_search_predicate_cost_upper_bound = 6
-    grammar_search_score_function = "hadd_lookahead_depth1"
+    grammar_search_score_function = "hff_lookahead_depth1"
```
Comment (tomsilver) [inline src/settings.py:122]: i'm still nervous about using hff with lookahead, but I guess we're going to immediately investigate this so it's fine

File: analysis/grammar_search_analysis.py [hunk 3]
```diff
@@ -179,12 +179,12 @@ def _main() -> None:
...
     score_function_names = [
         "prediction_error",
-        "hadd_lookahead",
+        "hadd_lookahead_depth0",
         "exact_lookahead",
         "hadd_lookahead_depth1",
         "hadd_lookahead_depth2",
     ]
-    run_planning = True
+    run_planning = False
```
Comment (tomsilver) [inline analysis/grammar_search_analysis.py:187]: is this intentional?

**Review by tomsilver — APPROVED** [review]

File: src/settings.py (hunk 2 above)
Comment (ronuchit) [reply src/settings.py:122]: yup, exactly. it's the only one that's doing the right thing in painting

**Pull request merged**